modules/analysis/src/model.rs

```diff
@@ -0,0 +1,142 @@
+use lazy_static::lazy_static;
+use parking_lot::RwLock;
+use petgraph::Graph;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::fmt;
+use std::sync::Arc;
+use trustify_entity::relationship::Relationship;
+use utoipa::ToSchema;
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct AnalysisStatus {
+    pub sbom_count: i32,
+    pub graph_count: i32,
+}
+impl fmt::Display for crate::model::AnalysisStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "graph_count {}", self.graph_count)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct PackageNode {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+}
+impl fmt::Display for PackageNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.purl)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct AncNode {
+    pub purl: String,
+    pub name: String,
+}
+impl fmt::Display for crate::model::AncNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.purl)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct AncestorSummary {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+    pub ancestors: Vec<AncNode>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct DepNode {
+    pub purl: String,
+    pub name: String,
+    pub deps: Vec<DepNode>,
+}
+impl fmt::Display for crate::model::DepNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.purl)
+    }
+}
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct DepSummary {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+    pub deps: Vec<DepNode>,
+}
+#[derive(Debug)]
+pub struct GraphMap {
+    map: HashMap<String, Graph<PackageNode, Relationship, petgraph::Directed>>,
+}
+
+// Static variables for the singleton instance
+lazy_static! {
```

Comment:
IIRC we had a discussion about this. And I also wouldn't see this as a blocker, but I think the following pattern is the replacement for using `lazy_static`:

```rust
static G: OnceLock<Arc<Mutex<usize>>> = OnceLock::new();

fn g() -> &'static Arc<Mutex<usize>> {
    G.get_or_init(|| Arc::new(Mutex::new(0)))
}
```